### DISCUSSION TITLE
在配置文件中可以设置全局并行任务数，但是如何设置单个工作流的并行度，现版本有这个功能吗？

### DISCUSSION THREAD

**Discussion by Length-Width-Height** [apache/dolphinscheduler#13588] [Q&A]
在一个工作流中有很多并行的task，我能对这个工作流设置允许的最大并行数么？不是全局的那种
![image](https://user-images.githubusercontent.com/117889161/220222646-f0a523f1-a33f-4149-b208-0fe9b112dddd.png)

**Comment by Tandoy** [comment]
As far as I know, the number of concurrent tasks for a single workflow has not been set yet. However, you can achieve this by adjusting the DAG.

**Reply by Length-Width-Height** [reply]
However, how to adjust the DAG to achieve the parallelism setting without affecting the original dependency?

**Reply by Tandoy** [reply]
You can make adjustments by referring to the following two pictures
![1](https://user-images.githubusercontent.com/56899730/222347595-d070a937-f0fc-47bc-a4bf-ae14fdce1b31.png)

![2](https://user-images.githubusercontent.com/56899730/222347606-e6fd7237-0389-419a-8f76-1248efe3f56d.png)

**Reply by Length-Width-Height** [reply]
![image](https://user-images.githubusercontent.com/117889161/222348419-b5aaca1d-1c8a-4b19-a146-428988cc7170.png)
Thanks for your reply. Take your diagram as an example. In the original DAG, when A fails to run, C and D can still run normally. However, with this new DAG, after A fails to run, all subsequent tasks will fail

**Reply by Tandoy** [reply]
Yes, there will be such problems. Perhaps you can also split the workflow in your problem into multiple workflows to run.And set the timing scheduling time to be consistent

![3](https://user-images.githubusercontent.com/56899730/222350428-8a58af34-34f1-4897-abae-dc5f959ef1bc.png)

**Comment by fuchanghai** [comment]
@Tandoy  @Length-Width-Height  can we use the ``` task group option ``` to reslove this question?

cc @ruanwenjun

**Reply by ruanwenjun** [reply]
Task group can support this.

**Comment by ruanwenjun** [comment]
Right now we cannot support define slot number for a workflow. This should be a good feature.
